Makefile

```diff
@@ -1,5 +1,8 @@
 DESTDIR=/usr/local
 PREFIX=mbedtls_
+override CFLAGS+=-std=gnu99
+MAKEOVERRIDES += CFLAGS+=-std=gnu99
```

Comment:
Is this ok with ARM compiler?

Reply:
No (CI confirms), and it's not ok with many other compilers either. Our makefile should be compiler-agnostic, at least if you pass a different `WARNING_CFLAGS` argument. I can't think of a case where an `override` in the makefile would be acceptable: when it comes to compiler flags, the makefile suggests, but the user decides.

Reply:
If `_GNU_SOURCE` or some such is needed to support a library feature, the `#define` needs to go in the appropriate `.c` or `.h` file. If the code is using a GNU extension to the C language, it shouldn't, or if the feature is optional (e.g. extra safety or optimization), it should be guarded by conditional compilation.

Reply:
`-std=c99` and `-std=gnu99` are made by `C_EXTENSIONS` in cmake.
and `CMAKE_C_EXTENSIONS` is `ON` default.
so we need set it to `OFF` when cmake?

we should set different flag for different compiler also.
but I have no idea about how do it in Makefile

Reply:
Passing `-std=c99` when you know that the compiler supports it (GCC, Clang, maybe other compilers with a gcc-compatible command line) is ok. But don't pass it if you don't know. And don't pass `-std=gnu99`: our code shouldn't require any GNU extension.

The makefiles are designed to be portable as long as you have GNU make. At least `make lib` should work on any OS with any compiler that accepts the options `-o` and `-c`. They don't even assume Unix utilities, and targets like `make clean` that assume Unix utilities (such as `rm`) have a Windows alternative (`del`). I don't think it's realistic to check the compiler in the makefiles. Note that cross-compilation is very common.

It's ok if some compilers require passing different flags when compiling, for example passing `make WARNING_CFLAGS="-Wno-something"` or `make CFLAGS="-mfoo -fno-weird-optimization"`. We expect the default to work on the platforms that people use the most (at least when not cross-compiling), but we know we can't cover them all. It's not ok if the makefile passes `-std=c99` to the compiler whether the user wanted it or not.

Reply:
so we shouldn't add c99 flag in Makefile?

Reply:
> so we shouldn't add c99 flag in Makefile?

Right. Unless you find a way to do so that does the right thing for compilers that don't support `-std=`, is OS-agnostic, and doesn't make people who aren't GNU make experts want to gouge their eyes out (GNU make is a difficult language once you get past the basics).

Reply:
I have revert changes in Makefile